modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCSSOConsentTestCase.java

```diff
@@ -0,0 +1,343 @@
+
+        authorizationCode = keyValues.get(0).getValue();
+        Assert.assertNotNull(authorizationCode, "Authorization code not received for " + application
+                .getApplicationName());
+        EntityUtils.consume(response.getEntity());
+    }
+
+    private void testGetAccessToken(OIDCApplication application) throws Exception {
+
+        HttpResponse response = sendGetAccessTokenPost(client, application);
+        Assert.assertNotNull(response, "Access token response is invalid for " + application.getApplicationName());
+        EntityUtils.consume(response.getEntity());
+
+        response = sendPostRequest(client, String.format(OIDCUtilTest.targetApplicationUrl, application.getApplicationContext() +
+                OAuth2Constant.PlaygroundAppPaths.appAuthorizePath));
+
+        Map<String, Integer> keyPositionMap = new HashMap<>(1);
+        keyPositionMap.put("name=\"accessToken\"", 1);
+        List<DataExtractUtil.KeyValue> keyValues = DataExtractUtil.extractInputValueFromResponse(response,
+                keyPositionMap);
+        Assert.assertNotNull(keyValues, "Access token not received for " + application.getApplicationName());
+
+        accessToken = keyValues.get(0).getValue();
+        Assert.assertNotNull(accessToken, "Access token not received for " + application.getApplicationName());
+        EntityUtils.consume(response.getEntity());
+
+        response = sendPostRequest(client, String.format(OIDCUtilTest.targetApplicationUrl, application.getApplicationContext() +
+                OAuth2Constant.PlaygroundAppPaths.appAuthorizePath));
+
+        keyPositionMap = new HashMap<>(1);
+        keyPositionMap.put("id=\"loggedUser\"", 1);
+        keyValues = DataExtractUtil.extractLabelValueFromResponse(response, keyPositionMap);
+        Assert.assertNotNull(keyValues, "No user logged in for " + application.getApplicationName());
+
+        String loggedUser = keyValues.get(0).getValue();
+        Assert.assertNotNull(loggedUser, "Logged user is null for " + application.getApplicationName());
+        EntityUtils.consume(response.getEntity());
+    }
+
+    private void testUserClaims() throws Exception {
+
+        HttpGet request = new HttpGet(OAuth2Constant.USER_INFO_ENDPOINT);
+
+        request.setHeader("User-Agent", OAuth2Constant.USER_AGENT);
+        request.setHeader("Authorization", "Bearer " + accessToken);
+        request.setHeader("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8");
+
+        HttpResponse response = client.execute(request);
+        BufferedReader rd = new BufferedReader(new InputStreamReader(response.getEntity().getContent()));
+
+        Object obj = JSONValue.parse(rd);
+        String claim = ((org.json.simple.JSONObject) obj).get("email").toString();
+
+        EntityUtils.consume(response.getEntity());
+        Assert.assertEquals("admin@wso2.com", claim, "Incorrect claim value");
```

Comment:
I guess this 'admin@wso2.com', user comes here based on the access token used in the method. If so it's better to introduce a constant for this user if there isn't one already. This is to avoid any failures in the future if someone changed the username in the token retrieval method.

Reply:
Removed this method since the claims that are asked for consent, are already tested and this is not required to test this in this testcase.